pkg/gcv/result.go

```diff
@@ -198,16 +199,22 @@ func (cv *ConstraintViolation) name() string {
 func (cv *ConstraintViolation) toViolation(name string, ancestryPath string) (*validator.Violation, error) {
 	metadataJson, err := json.Marshal(cv.metadata(map[string]interface{}{ancestryPathKey: ancestryPath}))
 	if err != nil {
-		return nil, errors.Wrapf(
-			err, "failed to marshal result metadata %v to json", cv.Metadata)
+		return nil, errors.Wrapf(err, "failed to marshal result metadata %v to json", cv.Metadata)
 	}
 	metadata := &structpb.Value{}
 	if err := jsonpb.UnmarshalString(string(metadataJson), metadata); err != nil {
 		return nil, errors.Wrapf(err, "failed to unmarshal json %s into structpb", string(metadataJson))
 	}
 
+	pbVal, err := cf.ConvertToProtoVal(cv.Constraint.Object)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to convert constraint object into proto struct value")
+	}
+	constraintConfig := &validator.Constraint{Metadata: pbVal}
```

Comment:
Shouldn't metadata contain the value for `constraint.metadata` not the full constraint? https://github.com/forseti-security/config-validator/blob/6cce12f5cd0e089a5371277384d9379d17b38288/pkg/gcv/oldconfigs/config.go#L105

Reply:
It does, you're right.

I'm interested in storing and passing around the entire config. I am thinking I could either:
1. Add a new field "FullConfig" to the ConstraintConfig message that has the full config.
2. Remove ConstraintConfig as a type and just use structpb.Value{} everywhere.

What would you suggest?

Reply:
What part of the config do you need? I think we should add the additional parts of `Constraint` that you need onto `ConstraintConfig` (ex. Spec).

Reply:
I don't need any particular part, I'd just like to be able to display the full rule that triggered a violation.

Reply:
I've updated the PR to add a new field to `validator.proto` and to set that to the full config while setting the `metadata` field to just the metadata. Not sure if this the way, considering metadata is now duplicated, but it does avoid changing the expectations of anyone using this proto and accessing the `metadata` field.

Reply:
Sorry, but I'd still like to avoid changing the proto's intent too much. `ConstraintConfig` should mirror the constraint yaml so if you need all properties we should add `apiVersion`, `kind`, and `spec` to the proto.

Reply:
Done.